### PR TITLE
[android] Bring back `add business`

### PR DESCRIPTION
**Pull request by kirylkaveryn**
Bring back the `addBusiness` button that was removed by the https://github.com/organicmaps/organicmaps/pull/8752

image

### CODE REVIEW

**Comment by strump** [comment]
✅ Tested on Android 14 - add business button is back

**Review by strump — APPROVED** [review]

**Comment by biodranik** [comment]
As discussed with Kiryl, there should be no Add Business mention anywhere, as it was started in https://github.com/organicmaps/organicmaps/pull/8158

**Comment by kirylkaveryn** [comment]
@strump @biodranik As a result of the discussion I totally removed the `addBusiness` button from the Android and merge the business logic with the `addPlace` (because they actually have the same behaviour and text).

@strump can you please test the Android one more time and check that I removed the button correctly?

image
image
image

**Comment by strump** [comment]
Updated PR: removed `place_page_add_business` from UI completely. Removed handlers.

@biodranik we had two methods in `PlacePageView` class:

```java
private void addOrganisation()
{
  ((MwmActivity) requireActivity()).showPositionChooserForEditor(true, false);
}

private void addPlace()
{
  ((MwmActivity) requireActivity()).showPositionChooserForEditor(false, true);
}
```

I removed first one. Shoud I also remove `isBusiness` parameter from `MwmActivity.showPositionChooserForEditor(boolean isBusiness, boolean applyPosition)`? What does this flag do?

**Comment by kirylkaveryn** [comment]
The latest update contains:
1. the `Add Business` button is used for the buildings (with the text "Add Business" instead of "Add Place" - see reverted commit 9b9e26bf5924140854add3d3d830c68340bb8e8d) on the Android to support magnifying to the building boundary.
2. on iOS the old Add Place is used and should be improved

**Comment by biodranik** [comment]
@meenbeese @Kazing FYI, we're reverting back changes from https://github.com/organicmaps/organicmaps/pull/8158 and https://github.com/organicmaps/organicmaps/issues/4321 because _there is_ a difference between Add Business inside a building and Add Place outside. Add Business on Android doesn't allow you to choose locations outside of the building, helping to map infrastructure and entrances.

File: data/strings/strings.txt [hunk 1]
```diff
@@ -15794,6 +15794,50 @@
...
+    pt-BR = Adicionar uma empresa
+    ro = Adaugă o firmă
+    ru = Добавить организацию
+    sk = Pridať organizáciu
+    sv = Lägg till organisation
+    th = เพิ่มองค์กร
+    tr = Kuruluş ekle
+    uk = Додати організацію
+    vi = Thêm tổ chức
+    zh-Hans = 将地点添加到 OpenStreetMap
```
Comment (biodranik) [inline data/strings/strings.txt:15838]:
```suggestion
    zh-Hans = 添加组织
```

File: data/strings/strings.txt [hunk 2]
```diff
@@ -15794,6 +15794,50 @@
...
+    zh-Hant = 將地點添加到 OpenStreetMap
```
Comment (biodranik) [inline data/strings/strings.txt:15839]:
```suggestion
    zh-Hant = 添加組織
```

**Review by biodranik — APPROVED** [review]
As discussed, iOS fix will come in a different PR

For some reason, both Add Place and Add Business **do not center the screen on the selection**. Is it a regression? I've refactored the passed point parameter here, will it help to fix this issue? https://github.com/organicmaps/organicmaps/pull/9005

**Comment by kirylkaveryn** [comment]
> **do not center the screen on the selection**. Is it a regression?

I'll investigate is it regression or not

**Comment by biodranik** [comment]
@kirylkaveryn it works on iOS, but it could be broken for a long time on Android. Can you please double-check the build before changes in #8752 ?

**Comment by kirylkaveryn** [comment]
> @kirylkaveryn it works on iOS, but it could be broken for a long time on Android. Can you please double-check the build before changes in #8752 ?

I checked- It was broken before my PR 😕

**Review by biodranik — APPROVED** [review]
Thanks! Can you please squash all code commits into one, plus one string revert+fix commit, plus one string regen commit?

**Comment by kirylkaveryn** [comment]
@biodranik done!

**Review by biodranik — APPROVED** [review]
Thanks! The next step is iOS )

**Pull request merged**